### ISSUE TITLE
Yarn causing ember-cli build to fail

### ISSUE THREAD

**Issue by mansona** [ember-cli/ember-cli#7328]
Hi all, 

I mostly wanted to open up this issue as a discussion point, I don't have any reason to want yarn builds to work it's just to bring it to everyone's attention and have a signpost on the internet for anyone else having the same issues.

I have been helping to update ember-cli-rollbar to a later version of ember-cli et al. but after a rebase it just started failing because of a random error. Here is the PR: https://github.com/davewasmer/ember-cli-rollbar/pull/107 and here is the failing build on travis: https://travis-ci.org/davewasmer/ember-cli-rollbar/jobs/272901179 

My first thought was that maybe I left a merge conflict in the simple rebase so I tried it locally and there were no failures. The only difference that I could ascertain was that Travis was using yarn to install in ember-try and I was using npm@5 to install locally

After removing yarn from the build process it seems like it worked fine (following the work done here: https://github.com/ember-cli/ember-try/pull/138 )

From my perspective this is an open and shut case: doesn't work with yarn, rip yarn out it works fine 🎉  but I don't really know where to go from here. Should we investigate the semantic differences in this particular case to cause it to fail in one and not the other? Should we just encourage people to use npm@5 instead of yarn? 

Feel free to just close this issue if you think that it is off topic 👍

**Comment by BYK** [comment]
Yarn team here. Would love to help investigate the issue and resolve it if that's something on our end. Sincere apology upfront if it is us.

**Comment by mansona** [comment]
@BYK to be honest it looks to be something in the depths of broccoli and beyond my direct experience and available time to debug 😞 

If you want to spend some time investigating I would recommend trying to build this particular branch with yarn: https://github.com/stonecircle/ember-cli-rollbar/tree/bug/yarn-not-working and taking a look at the failing travis build I linked in the main part of the issue: https://travis-ci.org/davewasmer/ember-cli-rollbar/jobs/272901179

Like I said this seems quite obscure an issue, the only thing I know is that when I removed yarn it seemed to work. Or more specifically, when I manually added an updated ember-try that did not use yarn to install dependencies it started working, but you catch my drift 😄

**Comment by BYK** [comment]
@mansona I don't see a `.yarnlock` file in that branch. Is that intentional?

**Comment by mansona** [comment]
@BYK well yes it is intentional, I'm sorry but I'm not getting the impression that you are reading any of the material that I have been posting. If you take a look here https://github.com/ember-cli/ember-try/pull/138 you will see the reason why we have **unintentionally** hit into this issue. 

If you want to talk about this issue more please DM me on twitter https://twitter/real_ate and let's keep this issue free for the ember-cli team to jump in if they would like 👍

**Comment by BYK** [comment]
>@BYK well yes it is intentional, I'm sorry but I'm not getting the impression that you are reading any of the material that I have been posting. If you take a look here ember-cli/ember-try#138 you will see the reason why we have unintentionally hit into this issue.

I'm sorry you got that impression. _any_ seems a bit broad. I did read the material but I didn't follow the link since I wasn't interested in the specifics yet. My first focus was to see how and why yarn was failing.

>After removing yarn from the build process it seems like it worked fine (following the work done here: ember-cli/ember-try#138 )

This didn't immediately make it apparent for me that `yarn` was completely removed and I failed to look at this from a non-yarn person perspective, and expected a lock file when using yarn. Failing on my side, so apologies :)

>If you want to talk about this issue more please DM me on twitter https://twitter/real_ate and let's keep this issue free for the ember-cli team to jump in if they would like 👍

Will do, thanks!

**Comment by theseyi** [comment]
Just a note that it might be helpful to others to keep (/ eventually add) the discussions and possible resolution here incase we encounter a similar issue?

**Comment by mansona** [comment]
@theseyi I reached out to Burak on twitter to continue this discussion but he never got around to debugging this issue on his side. 

I was afraid that the conversation between us would derail the main point of what I was trying to say: this was supposed to be a discussion point so that we might think about backing away from supporting yarn out of the box and just "revert" to pure npm@5 support.

**Comment by Turbo87** [comment]
In a quick Slack call we determined that this was likely caused by the CI system using `npm install` to install the dependencies and `ember-try` (back then) auto-detecting the `yarn.lock` file and using `yarn install` for installing the variable dependencies which caused the `node_modules` folder to get into an inconsistent state.

`ember-try` has been fixed since then by dropping the auto-detection and requiring an explicit `useYarn` flag if `yarn install` is supposed to be used.

Feel free to comment and reopen if this is still an issue for you.
